File: core/src/test/java/tech/reinders/kotlin/databasemigrator/H2Connector.java
package tech.reinders.kotlin.databasemigrator;

import tech.reinders.kotlin.databasemigrator.util.JdbcUtil;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.fail;

public final class H2Connector {
    private static final String FILE = "/tmp/test.tmp";
    private static final String URL = "jdbc:h2:" + FILE;
    private static final String USR = "sa";
    private static final String PWD = "";

    private final List<Connection> connections = new ArrayList<>();

    public Connection getConnection() {
        try {
            final Connection connection = DriverManager.getConnection(URL, USR, PWD);
            this.connections.add(connection);
            return connection;
        } catch (SQLException e) {
            fail(e);
            throw new RuntimeException(e);
        }
    }

    public void close() {
        try {
            getConnection().createStatement().execute("DROP ALL OBJECTS");
        } catch (Exception ex) {
            throw new RuntimeException("Error while dropping all objects", ex);
        }

        connections.forEach(JdbcUtil.INSTANCE::close);

        new File(FILE).delete();
    }
}

File: core/src/test/java/tech/reinders/kotlin/databasemigrator/H2Connector.kt
package tech.reinders.kotlin.databasemigrator

import org.junit.jupiter.api.Assertions
import java.io.File
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException
import java.util.*
import java.util.function.Consumer

class H2Connector {
    private val connections: MutableList<Connection> = ArrayList()
    val connection: Connection
        get() = try {
            val connection = DriverManager.getConnection(URL, USR, PWD)
            connections.add(connection)
            connection
        } catch (e: SQLException) {
            Assertions.fail<Any>(e)
            throw RuntimeException(e)
        }

    fun close() {
        try {
            connection.createStatement().execute("DROP ALL OBJECTS")
        } catch (ex: Exception) {
            throw RuntimeException("Error while dropping all objects", ex)
        }
        connections.forEach(Consumer { obj: Connection -> obj.close() })
        File(FILE).delete()
    }

    companion object {
        private const val FILE = "/tmp/test.tmp"
        private const val URL = "jdbc:h2:$FILE"
        private const val USR = "sa"
        private const val PWD = ""
    }
}